### PR TITLE
add ExtendedEditValidator

### PR DESCRIPTION
**Pull request by ndoriomedov**
http://outsourcing.nat.tepkom.ru:20080/redmine/issues/975

### CODE REVIEW

**Review by MariyaSin — APPROVED** [review]

**Review by asyagraf — APPROVED** [review]

File: src/Kernel/Validators/ExtendedEditRequestValidator.cs [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    protected void AddСorrectPaths(List<string> paths)
+    {
+      if (paths.FirstOrDefault(p => p.Equals(RequestedOperation.path[1..], StringComparison.OrdinalIgnoreCase)) == null)
+      {
+        Context.AddFailure(RequestedOperation.path, $"This path {RequestedOperation.path} is not available");
+      }
+    }
+
+    protected void AddСorrectOperations(
+        string propertyName,
```
Comment (VladSpitsin) [inline src/Kernel/Validators/ExtendedEditRequestValidator.cs]: fix tabs

File: src/Kernel/Validators/ExtendedEditRequestValidator.cs [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+        List<OperationType> types)
+    {
+      if (RequestedOperation.path.EndsWith(propertyName, StringComparison.OrdinalIgnoreCase)
+          && !types.Contains(RequestedOperation.OperationType))
```
Comment (VladSpitsin) [inline src/Kernel/Validators/ExtendedEditRequestValidator.cs]: fix tabs

File: src/Kernel/Validators/ExtendedEditRequestValidator.cs [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+      {
+        Context.AddFailure(propertyName, $"This operation {RequestedOperation.OperationType} is prohibited for {propertyName}");
+      }
+    }
+
+    protected void AddFailureForPropertyIf(
+        string propertyName,
```
Comment (VladSpitsin) [inline src/Kernel/Validators/ExtendedEditRequestValidator.cs]: same

File: src/Kernel/Validators/ExtendedEditRequestValidator.cs [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+        Func<OperationType, bool> type,
+        Dictionary<Func<Operation<T>, bool>, string> predicates,
+        CascadeMode mode = CascadeMode.Continue)
+    {
+      if (!RequestedOperation.path.EndsWith(propertyName, StringComparison.OrdinalIgnoreCase)
+          || !type(RequestedOperation.OperationType))
```
Comment (VladSpitsin) [inline src/Kernel/Validators/ExtendedEditRequestValidator.cs]: same

File: src/Kernel/Validators/ExtendedEditRequestValidator.cs [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+          if (mode != CascadeMode.Continue)
+          {
+            break;
+          }
+        }
+      }
+    }
+
+    protected async Task AddFailureForPropertyIfAsync(
+        string propertyName,
```
Comment (VladSpitsin) [inline src/Kernel/Validators/ExtendedEditRequestValidator.cs]: tabs

File: src/Kernel/Validators/ExtendedEditRequestValidator.cs [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+        Func<OperationType, bool> type,
+        Dictionary<Func<Operation<T>, Task<bool>>, string> predicates,
+        CascadeMode mode = CascadeMode.Continue)
+    {
+      if (!RequestedOperation.path.EndsWith(propertyName, StringComparison.OrdinalIgnoreCase)
+          || !type(RequestedOperation.OperationType))
```
Comment (VladSpitsin) [inline src/Kernel/Validators/ExtendedEditRequestValidator.cs]: same

**Review by VladSpitsin — APPROVED** [review]

**Review by borovHokage — APPROVED** [review]

**Pull request merged**